kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/KieServicesClientImpl.java

```diff
@@ -123,6 +119,30 @@ private void init() {
 
     }
 
+    private List<String> getCapabilitiesFromServer() {
+        if (config.isJms()) {
+            if (config.isJmsTransactional()) {
+                kieServerInfo = getServerInfoInTransaction();
```

Comment:
the use case for transactional jms is to send message to server only when transaction completes and in async way as there won't be response until tx completes.
Moreover for async execution expected server capabilities needs to be provided as they can't be loaded automatically since the response will arrive later on. 

KieServicesClients need to be tx dependencies free as it might or might not run within tx context so it does not need to do any manual look ups itself. Thus the setting to make the jms delivery transactional so when ConnectionFactory is XA/JTA then it can deliver message only if transaction commits.